spec/rubocop/options_spec.rb

```diff
@@ -262,4 +262,44 @@ def abs(path)
       end
     end
   end
+
+  describe 'options from file' do
+    let(:command_line_options) { [] }
+
+    before do
+      create_file('.rubocop', %w(--no-color --fail-fast))
+    end
+
+    subject { options.parse(command_line_options).first }
+
+    context 'no command line options given' do
+      it 'reads options from the file' do
+        is_expected.to eq(color: false, fail_fast: true)
+      end
+    end
+
+    context 'command line options given' do
+      let(:command_line_options) { ['--color'] }
+
+      it 'CLI options have priority over file options' do
+        is_expected.to eq(color: true, fail_fast: true)
+      end
+    end
+  end
+
+  describe 'options from RUBOCOP_OPTS environment variable' do
+    let(:command_line_options) { %w(--no-color --fail-fast) }
+
+    around do |example|
+      ENV['RUBOCOP_OPTS'] = '--color'
+      example.run
+      ENV.delete('RUBOCOP_OPTS')
+    end
+
+    subject { options.parse(command_line_options).first }
+
+    it 'ENV options have priority over command line options' do
```

Comment:
It breaks the whole idea. You can put `.rubocop` file in the repository, so every one in the team will get reasonable set of default options. If one need additional options, she can pass command line options and override `.rubocop` settings.

Another example, I have a Makefile for running various tasks in docker. So I can run specs, rubocop, etc. in isolated and reproducible environment. Its very helpful to pass special options in some environments. 

When I run RSpec on CI server, I prefer to use JUnit formatter and redirect output to file, so CI server can read this file and generate pretty output. This is done by configuring CI server, to set environment variable `SPEC_OPTS="--format RspecJunitFormatter --out rspec.xml"`.  

Thus, I can decouple my build script from knowledge about specific CI servers.

So, the purpose of `.rubocop` file is to provide good defaults, but the purpose of `RUBOCOP_OPTS` is to externally override options. 

For instance, `.rubocop` file may contains `-D` option, and RUBOCOP_OPTS may be set to `--require rubocop/formatter/checkstyle_formatter --format RuboCop::Formatter::CheckstyleFormatter --no-color -out rubocop.xml`

So I think that order of precedence should:

1.` RUBOCOP_OPTS` (Highest)
2. Command line options
3. `.rubocop` file (lowest)